dev/com.ibm.ws.security.wim.adapter.ldap/src/com/ibm/ws/security/wim/adapter/ldap/LdapAdapter.java

```diff
@@ -534,27 +534,32 @@ public Root login(Root root) throws WIMException {
             }
 
             String searchExpr = "@xsi:type=" + quote + qName + quote + " and "
-                                + SchemaConstants.PROP_PRINCIPAL_NAME + "=" + quote + principalName + quote;
+                            + SchemaConstants.PROP_PRINCIPAL_NAME + "=" + quote + principalName + quote;
 
             loginCtrl.setExpression(searchExpr);
             srchCtrl = getLdapSearchControl(loginCtrl, false, false);
             String[] searchBases = srchCtrl.getBases();
-            String sFilter = srchCtrl.getFilter();
+            String sFilter = null;
 
-            if (iLdapConfigMgr.getUseEncodingInSearchExpression() != null)
-                sFilter = LdapHelper.encodeAttribute(sFilter, iLdapConfigMgr.getUseEncodingInSearchExpression());
-
-            //Check is input principalName is DN, if it is DN use the same for login otherwise use userFilter
-            String principalNameDN = null;
-            try {
-                principalNameDN = new LdapName(principalName).toString();
-            } catch (InvalidNameException e) {
-                e.getMessage();
-            }
-            Filter userFilter = iLdapConfigMgr.getUserFilter();
-            if (principalNameDN == null && userFilter != null) {
-                sFilter = userFilter.prepare(principalName);
-                sFilter = setAttributeNamesInFilter(sFilter, SchemaConstants.DO_PERSON_ACCOUNT);
+            if (!iLdapConfigMgr.loginPropertyDefined()) {
+                //Check is input principalName is DN, if it is DN use the same for login otherwise use userFilter
+                String principalNameDN = null;
+                try {
+                    principalNameDN = new LdapName(principalName).toString();
+                } catch (InvalidNameException e) {
+                    e.getMessage();
+                }
+                Filter userFilter = iLdapConfigMgr.getUserFilter();
+                if (principalNameDN == null && userFilter != null) {
+                    sFilter = userFilter.prepare(principalName);
+                    sFilter = setAttributeNamesInFilter(sFilter, SchemaConstants.DO_PERSON_ACCOUNT);
+                }
+            }
+            if (sFilter == null) {
+                sFilter = srchCtrl.getFilter();
+                if (iLdapConfigMgr.getUseEncodingInSearchExpression() != null) {
```

Comment:
Does this not apply if userFilter is used instead?

Reply:
(I'm nots sure what this call does)

Reply:
srchCtrl.getFilter does not apply if the userFilter is used. However, the userFilter is only used if loginProperties are not defined. I changed the order so that we aren't calling srchCtrl.getFilter when we aren't using it.

Reply:
I meant the getUseEncodingInSearchExpression() and encodeAttribute(). Was that a stand-alone LDAP setting? Should we be checking whether to encode attributes whether the filter is statically or dynamically defined? The answer may be "no", but I am not sure.

Reply:
As discussed in slack, this is a property we have not exposed. I will handle this in an additional work item.